src/Bundle/Doctrine/ORM/ExpressionBuilder.php

```diff
@@ -27,188 +27,135 @@ public function __construct(QueryBuilder $queryBuilder)
         $this->queryBuilder = $queryBuilder;
     }
 
-    /**
-     * {@inheritdoc}
-     */
     public function andX(...$expressions)
     {
         return $this->queryBuilder->expr()->andX(...$expressions);
     }
 
-    /**
-     * {@inheritdoc}
-     */
     public function orX(...$expressions)
     {
         return $this->queryBuilder->expr()->orX(...$expressions);
     }
 
-    /**
-     * {@inheritdoc}
-     */
     public function comparison(string $field, string $operator, $value)
     {
         return new Comparison($field, $operator, $value);
     }
 
-    /**
-     * {@inheritdoc}
-     */
     public function equals(string $field, $value)
     {
-        $rootAlias = $this->queryBuilder->getRootAlias();
-        if (0 === strpos($field, $rootAlias . '.')) {
-            $field = substr_replace($field, '', 0, strlen($rootAlias) + 1);
-        }
-
+        $field = $this->adjustField($field);
         $parameterName = $this->getParameterName($field);
```

Comment:
Are we sure that we want to run `$this->getParameterName()` with **adjusted field** as an argument? Isn't it changing the previous behaviour and breaking BC?

Reply:
`adjustField` is a method that coheres the given field, because field could look like `o.order` or `order`. So yes, the name returned by `getParameterName` could be now different (e.q. `order` instead of `o_order`), because I always remove the root alias, but I don't think that this is a BC break. The same grid configuration will work in the same way as it was before.